app/controllers/departments_controller.ts

```diff
@@ -0,0 +1,47 @@
+import type { HttpContext } from "@adonisjs/core/http";
+
+import Department from "#models/department";
+import { paginationValidator } from "#validators/pagination";
+import { showValidator } from "#validators/show";
+
+export default class DepartmentsController {
+  protected readonly relations = ["fieldOfStudy", "departmentLink"];
+
+  async index({ request }: HttpContext) {
+    const { page, limit } = await request.validateUsing(paginationValidator);
+    const baseQuery = Department.query().withScopes((scopes) => {
+      scopes.handleSearchQuery(
+        request.except([
+          "logo",
+          "description",
+          "gradientStart",
+          "gradientStop",
+        ]),
+      );
```

Comment:
```suggestion
      scopes.handleSearchQuery(
        request.qs(),
        "logo",
        "description",
        "gradientStart",
        "gradientStop",
      );
```
\+ prettier cuz i typed this out by hand